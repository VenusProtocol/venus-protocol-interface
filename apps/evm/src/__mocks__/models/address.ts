const address = '0x3d759121234cd36F8124C21aFe1c6852d2bEd848';

export const altAddress = '0xa258a693A403b7e98fd05EE9e1558C760308cFC7';

export default address;
