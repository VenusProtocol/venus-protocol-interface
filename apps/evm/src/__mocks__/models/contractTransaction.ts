import { ChainId } from 'types';
import type { Transaction } from 'viem';

const contractTransaction: Transaction = {
  hash: '0xfake-transaction-hash',
  type: 'legacy',
  from: '0x1d759121234cd36F8124C21aFe1c6852d2bEd848',
  to: '0x2d759121234cd36F8124C21aFe1c6852d2bEd834',
  nonce: 10,
  value: 0n,
  chainId: ChainId.BSC_TESTNET,
  blockHash: '0xfake-block-hash',
  blockNumber: 1123213n,
  gas: 12n,
  gasPrice: 1n,
  input: '0xfake-input',
  r: '0xfake-r',
  s: '0xfake-s',
  v: 3n,
  transactionIndex: 1,
  typeHex: '0xfake-type-hex',
};

export default contractTransaction;
