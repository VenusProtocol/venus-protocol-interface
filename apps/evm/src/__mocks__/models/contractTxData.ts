import type { ContractTxData } from 'types';

const contractTxData: ContractTxData<any, any> = {
  contract: {},
  methodName: 'fakeMethod',
  args: [],
};

export default contractTxData;
