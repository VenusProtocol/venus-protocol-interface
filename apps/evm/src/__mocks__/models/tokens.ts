import bnbLogo from 'packages/tokens/img/bnb.svg';
import busdLogo from 'packages/tokens/img/busd.svg';
import ethLogo from 'packages/tokens/img/eth.svg';
import lisUsdLogo from 'packages/tokens/img/lisUSD.png';
import lunaLogo from 'packages/tokens/img/luna.svg';
import usdcLogo from 'packages/tokens/img/usdc.svg';
import usdtLogo from 'packages/tokens/img/usdt.svg';
import ustLogo from 'packages/tokens/img/ust.svg';
import vaiLogo from 'packages/tokens/img/vai.svg';
import vrtLogo from 'packages/tokens/img/vrt.svg';
import wbnbLogo from 'packages/tokens/img/wbnb.svg';
import xvsLogo from 'packages/tokens/img/xvs.svg';
import { Token } from 'types';

export const xvs: Token = {
  address: '0xB9e0E753630434d7863528cc73CB7AC638a7c8ff',
  decimals: 18,
  symbol: 'XVS',
  asset: xvsLogo,
};

export const vai: Token = {
  address: '0x5fFbE5302BadED40941A403228E6AD03f93752d9',
  decimals: 18,
  symbol: 'VAI',
  asset: vaiLogo,
};

export const vrt: Token = {
  address: '0x331F639B4F3CF6E391CD244e0b5027C5968Ec325',
  decimals: 18,
  symbol: 'VRT',
  asset: vrtLogo,
};

export const bnb: Token = {
  address: '0x0000000000000000000000000000000000000000',
  isNative: true,
  decimals: 18,
  symbol: 'BNB',
  asset: bnbLogo,
};

export const usdc: Token = {
  address: '0x16227D60f7a0e586C66B005219dfc887D13C9531',
  decimals: 6,
  symbol: 'USDC',
  asset: usdcLogo,
};

export const usdt: Token = {
  address: '0xA11c8D9DC9b66E209Ef60F0C8D969D3CD988782c',
  decimals: 6,
  symbol: 'USDT',
  asset: usdtLogo,
};

export const busd: Token = {
  address: '0x8301F2213c0eeD49a7E28Ae4c3e91722919B8B47',
  decimals: 18,
  symbol: 'BUSD',
  asset: busdLogo,
};

export const ust: Token = {
  address: '0x5A79efD958432E72211ee73D5DDFa9bc8f248b5F',
  decimals: 18,
  symbol: 'UST',
  asset: ustLogo,
};

export const luna: Token = {
  address: '0xf36160EC62E3B191EA375dadfe465E8Fa1F8CabB',
  decimals: 6,
  symbol: 'LUNA',
  asset: lunaLogo,
};

export const wbnb: Token = {
  symbol: 'WBNB',
  decimals: 18,
  address: '0xae13d989daC2f0dEbFf460aC112a837C89BAa7cd',
  asset: wbnbLogo,
};

export const eth: Token = {
  address: '0x98f7A83361F7Ac8765CcEBAB1425da6b341958a7',
  decimals: 18,
  symbol: 'ETH',
  asset: ethLogo,
};

export const lisUsd: Token = {
  address: '0xe73774DfCD551BF75650772dC2cC56a2B6323453',
  decimals: 18,
  symbol: 'lisUSD',
  asset: lisUsdLogo,
};

export default [xvs, bnb, usdc, usdt, busd, ust, luna, vai, wbnb, eth, vrt, lisUsd];
