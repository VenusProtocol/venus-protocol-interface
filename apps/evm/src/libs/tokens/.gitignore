# Automatically generated PancakeSwap token list
/generated