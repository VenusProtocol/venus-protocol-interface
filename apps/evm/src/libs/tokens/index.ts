export * from './infos';
export * from './utilities/getTokens';
export * from './utilities/getToken';
export * from './utilities/getSwapTokens';
export * from './utilities/isTokenActionEnabled';
export * from './hooks/useGetToken';
export * from './hooks/useGetTokens';
export * from './hooks/useGetSwapTokens';
