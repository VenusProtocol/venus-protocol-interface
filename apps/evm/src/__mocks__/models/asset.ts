import BigNumber from 'bignumber.js';

import { busd, usdc, usdt, xvs } from '__mocks__/models/tokens';
import { vBusd, vUsdc, vUsdt, vXvs } from '__mocks__/models/vTokens';

import { Asset } from 'types';

export const assetData: Asset[] = [
  {
    vToken: vXvs,
    supplyApyPercentage: new BigNumber('0.05225450324405023'),
    borrowApyPercentage: new BigNumber('-2.3062487835658776'),
    collateralFactor: 0.5,
    reserveFactor: 0.25,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('127.86734'),
    liquidityCents: new BigNumber(8036465875),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(49.589181233),
    userWalletBalanceTokens: new BigNumber('100'),
    userWalletBalanceCents: new BigNumber('12786'),
    userSupplyBalanceTokens: new BigNumber('90'),
    userSupplyBalanceCents: new BigNumber('11508'),
    userBorrowBalanceTokens: new BigNumber('0'),
    userBorrowBalanceCents: new BigNumber('0'),
    isCollateralOfUser: true,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(70925716),
    supplyBalanceCents: new BigNumber(278311516),
    supplyBalanceTokens: new BigNumber('19339683254955736'),
    borrowBalanceTokens: new BigNumber('1852935.597521220541385584'),
    supplyPercentageRatePerBlock: new BigNumber('0.000001'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000001'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.11720675342484096'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('4.17469243006608279'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
    ],
  },
  {
    vToken: vUsdc,
    supplyApyPercentage: new BigNumber('3.887242555711379188'),
    borrowApyPercentage: new BigNumber('-5.361233028654066'),
    collateralFactor: 0.8,
    reserveFactor: 0.2,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('99.99364'),
    liquidityCents: new BigNumber(1702951959),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(1),
    userWalletBalanceTokens: new BigNumber('0'),
    userWalletBalanceCents: new BigNumber('0'),
    userSupplyBalanceTokens: new BigNumber('100'),
    userSupplyBalanceCents: new BigNumber('9999'),
    userBorrowBalanceTokens: new BigNumber('0'),
    userBorrowBalanceCents: new BigNumber('0'),
    isCollateralOfUser: false,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(858721657509436),
    supplyBalanceCents: new BigNumber(1000183891879506),
    supplyBalanceTokens: new BigNumber('47171999131879271527200010830'),
    borrowBalanceTokens: new BigNumber('73128318.509651061457900627'),
    supplyPercentageRatePerBlock: new BigNumber('0.000002'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000002'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('1.353105649796123742'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdc,
        apyPercentage: new BigNumber('0.953105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: usdc,
        apyPercentage: new BigNumber('0.753105649796123742'),
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('1.670327607690572731'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdc,
        apyPercentage: new BigNumber('1.233105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: usdc,
        apyPercentage: new BigNumber('0.913105649796123742'),
      },
    ],
  },
  {
    vToken: vUsdt,
    supplyApyPercentage: new BigNumber('3.593608909332766999'),
    borrowApyPercentage: new BigNumber('-4.9748661428011145'),
    collateralFactor: 0.8,
    reserveFactor: 0.2,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('100.024602'),
    liquidityCents: new BigNumber(5534102886),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(0.981982),
    userWalletBalanceTokens: new BigNumber('900'),
    userWalletBalanceCents: new BigNumber('90000'),
    userSupplyBalanceTokens: new BigNumber('0'),
    userSupplyBalanceCents: new BigNumber('0'),
    userBorrowBalanceTokens: new BigNumber('40'),
    userBorrowBalanceCents: new BigNumber('4000'),
    isCollateralOfUser: false,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(3158444721),
    supplyBalanceCents: new BigNumber(1098041201011568),
    supplyBalanceTokens: new BigNumber('5029972090817266864401527367893625'),
    borrowBalanceTokens: new BigNumber('232511166.920938849475104194'),
    supplyPercentageRatePerBlock: new BigNumber('0.000003'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000003'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.421719501189155143'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdt,
        apyPercentage: new BigNumber('1.753105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.522209972682294832'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdt,
        apyPercentage: new BigNumber('1.015310564979612374'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
    ],
  },
  {
    vToken: vBusd,
    supplyApyPercentage: new BigNumber('2.886396363044176106'),
    borrowApyPercentage: new BigNumber('-4.050271277344538'),
    collateralFactor: 0.8,
    reserveFactor: 0.2,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('100.000922'),
    liquidityCents: new BigNumber(3654492935),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(1.000003),
    userWalletBalanceTokens: new BigNumber('110'),
    userWalletBalanceCents: new BigNumber('11000'),
    userSupplyBalanceTokens: new BigNumber('0'),
    userSupplyBalanceCents: new BigNumber('0'),
    userBorrowBalanceTokens: new BigNumber('50'),
    userBorrowBalanceCents: new BigNumber('5000'),
    isCollateralOfUser: false,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(83910350502),
    supplyBalanceCents: new BigNumber(1054707853878),
    supplyBalanceTokens: new BigNumber('51881081291203672464'),
    borrowBalanceTokens: new BigNumber('142662020.229587308931217432'),
    supplyPercentageRatePerBlock: new BigNumber('0.000004'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000004'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.678420831753642169'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: busd,
        apyPercentage: new BigNumber('1.753105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: xvs,
        apyPercentage: new BigNumber(0),
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.852697602175970714'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: busd,
        apyPercentage: new BigNumber('1.753105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: busd,
        apyPercentage: new BigNumber('0.913105649796123742'),
      },
    ],
  },
];
