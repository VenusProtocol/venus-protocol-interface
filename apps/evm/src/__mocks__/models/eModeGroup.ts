import type { Asset, EModeGroup } from 'types';
import { assetData } from './asset';

const generateEModeGroup = ({
  groupAssets,
  id,
  name,
}: { groupAssets: Asset[]; id: number; name: string }): EModeGroup => ({
  id,
  name,
  assetSettings: groupAssets.map(a => ({
    vToken: a.vToken,
    collateralFactor: a.userCollateralFactor + 0.1,
    liquidationThresholdPercentage: a.userLiquidationThresholdPercentage + 12,
    liquidationPenaltyPercentage: a.userLiquidationThresholdPercentage - 50,
    liquidityCents: a.liquidityCents.toNumber(),
    liquidityTokens: a.cashTokens,
    isBorrowable: a.isBorrowable,
  })),
});

export const eModeGroups: EModeGroup[] = [
  generateEModeGroup({
    id: 0,
    name: 'Stablecoins',
    groupAssets: assetData.slice(0, 3),
  }),
  generateEModeGroup({
    id: 1,
    name: 'DeFi',
    groupAssets: assetData.slice(2, 4),
  }),
  generateEModeGroup({
    id: 2,
    name: '#ToTheMoon',
    groupAssets: assetData.slice(5, 8),
  }),
];
