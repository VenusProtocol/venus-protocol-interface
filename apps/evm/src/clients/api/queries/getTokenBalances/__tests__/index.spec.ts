import type { PublicClient } from 'viem';

import fakeAccountAddress from '__mocks__/models/address';
import { bnb, busd, lisUsd } from '__mocks__/models/tokens';

import { getTokenBalances } from '..';

const tokens = [busd, lisUsd];

describe('getTokenBalances', () => {
  it('returns token balances in the right format on success', async () => {
    const fakePublicClient = {
      readContract: () => 1000000000000000000n,
    } as unknown as PublicClient;

    const res = await getTokenBalances({
      publicClient: fakePublicClient,
      accountAddress: fakeAccountAddress,
      tokens,
    });

    expect(res).toMatchSnapshot();
  });

  it('returns token balances, including native token, in the right format on success', async () => {
    const fakePublicClient = {
      readContract: () => 1000000000000000000n,
      getBalance: () => 1000000000000000000n,
    } as unknown as PublicClient;

    const res = await getTokenBalances({
      publicClient: fakePublicClient,
      accountAddress: fakeAccountAddress,
      tokens: [...tokens, bnb],
    });

    expect(res).toMatchSnapshot();
  });
});
