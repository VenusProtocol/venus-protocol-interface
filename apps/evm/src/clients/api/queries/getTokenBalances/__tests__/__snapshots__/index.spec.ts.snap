// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getTokenBalances > returns token balances in the right format on success 1`] = `
{
  "tokenBalances": [
    {
      "balanceMantissa": "1000000000000000000",
      "token": {
        "address": "0x8301F2213c0eeD49a7E28Ae4c3e91722919B8B47",
        "asset": "data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='24'%20height='24'%20fill='none'%20xmlns:v='https://vecta.io/nano'%3e%3cg%20clip-path='url(%23A)'%3e%3ccircle%20cx='12'%20cy='12'%20r='12'%20fill='%23f0b90b'/%3e%3cg%20fill='%23fff'%3e%3cpath%20d='M11.971%202.4l2.372%202.429L8.371%2010.8%206%208.429%2011.971%202.4zm3.6%203.6l2.372%202.429L8.371%2018%206%2015.629%2015.571%206zm-10.8%203.6l2.371%202.429L4.771%2014.4%202.4%2012.029%204.771%209.6zm14.401%200l2.371%202.429-9.571%209.572L9.6%2019.229%2019.172%209.6z'/%3e%3c/g%3e%3c/g%3e%3cdefs%3e%3cclipPath%20id='A'%3e%3cpath%20fill='%23fff'%20d='M0%200h24v24H0z'/%3e%3c/clipPath%3e%3c/defs%3e%3c/svg%3e",
        "decimals": 18,
        "symbol": "BUSD",
      },
    },
    {
      "balanceMantissa": "1000000000000000000",
      "token": {
        "address": "0xe73774DfCD551BF75650772dC2cC56a2B6323453",
        "asset": "/src/libs/tokens/img/underlyingTokens/lisUSD.png",
        "decimals": 18,
        "symbol": "lisUSD",
      },
    },
  ],
}
`;

exports[`getTokenBalances > returns token balances, including native token, in the right format on success 1`] = `
{
  "tokenBalances": [
    {
      "balanceMantissa": "1000000000000000000",
      "token": {
        "address": "0x8301F2213c0eeD49a7E28Ae4c3e91722919B8B47",
        "asset": "data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='24'%20height='24'%20fill='none'%20xmlns:v='https://vecta.io/nano'%3e%3cg%20clip-path='url(%23A)'%3e%3ccircle%20cx='12'%20cy='12'%20r='12'%20fill='%23f0b90b'/%3e%3cg%20fill='%23fff'%3e%3cpath%20d='M11.971%202.4l2.372%202.429L8.371%2010.8%206%208.429%2011.971%202.4zm3.6%203.6l2.372%202.429L8.371%2018%206%2015.629%2015.571%206zm-10.8%203.6l2.371%202.429L4.771%2014.4%202.4%2012.029%204.771%209.6zm14.401%200l2.371%202.429-9.571%209.572L9.6%2019.229%2019.172%209.6z'/%3e%3c/g%3e%3c/g%3e%3cdefs%3e%3cclipPath%20id='A'%3e%3cpath%20fill='%23fff'%20d='M0%200h24v24H0z'/%3e%3c/clipPath%3e%3c/defs%3e%3c/svg%3e",
        "decimals": 18,
        "symbol": "BUSD",
      },
    },
    {
      "balanceMantissa": "1000000000000000000",
      "token": {
        "address": "0xe73774DfCD551BF75650772dC2cC56a2B6323453",
        "asset": "/src/libs/tokens/img/underlyingTokens/lisUSD.png",
        "decimals": 18,
        "symbol": "lisUSD",
      },
    },
    {
      "balanceMantissa": "1000000000000000000",
      "token": {
        "address": "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
        "asset": "data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='24'%20height='24'%20fill='none'%20xmlns:v='https://vecta.io/nano'%3e%3cg%20clip-path='url(%23A)'%3e%3cg%20clip-path='url(%23B)'%3e%3cpath%20fill-rule='evenodd'%20d='M12%200a12%2012%200%201%201%200%2024%2012%2012%200%201%201%200-24z'%20fill='%23f0b90b'/%3e%3cg%20fill='%23fff'%3e%3cpath%20d='M6.595%2012l.009%203.173L9.3%2016.76v1.858l-4.274-2.507v-5.039L6.595%2012zm0-3.173v1.849l-1.57-.929V7.898l1.57-.929%201.578.929-1.578.929zm3.831-.929l1.57-.929%201.578.929-1.578.929-1.57-.929zM7.73%2014.515v-1.858l1.57.929v1.849l-1.57-.92zm2.696%202.91l1.57.929%201.578-.929v1.849l-1.578.929-1.57-.929v-1.849zm5.4-9.527l1.57-.929%201.578.929v1.849l-1.578.929V8.827l-1.57-.929zm1.57%207.275L17.405%2012l1.57-.929v5.038l-4.274%202.507v-1.858l2.695-1.586zm-1.126-.658l-1.57.92v-1.849l1.57-.929v1.858zm0-5.03l.009%201.858-2.704%201.587v3.181l-1.57.92-1.57-.92V12.93l-2.704-1.587V9.485l1.577-.929%202.687%201.594%202.704-1.594%201.578.929h-.007zM7.73%206.313l4.266-2.515%204.274%202.515-1.57.929-2.704-1.594L9.3%207.241l-1.57-.929z'/%3e%3c/g%3e%3c/g%3e%3c/g%3e%3cdefs%3e%3cclipPath%20id='A'%3e%3cpath%20fill='%23fff'%20d='M0%200h24v24H0z'/%3e%3c/clipPath%3e%3cclipPath%20id='B'%3e%3cpath%20fill='%23fff'%20d='M0%200h24v24H0z'/%3e%3c/clipPath%3e%3c/defs%3e%3c/svg%3e",
        "decimals": 18,
        "isNative": true,
        "symbol": "BNB",
      },
    },
  ],
}
`;
